plugins/by-name/kulala/default.nix

```
@@ -0,0 +1,190 @@
+{ lib, ... }:
+let
+  inherit (lib.nixvim) defaultNullOpts;
+  inherit (lib) types;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "kulala";
+  originalName = "kulala.nvim";
+  package = "kulala-nvim";
+
+  maintainers = [ lib.maintainers.BoneyPatel ];
+
+  settingsOptions = {
+    curl_path = defaultNullOpts.mkStr "curl" ''
+      Path to the cURL binary. Set this if cURL is installed in a non-standard location.
+    '';
+
+    display_mode =
+      defaultNullOpts.mkEnumFirstDefault
+        [
+          "split"
+          "float"
+        ]
+        ''
+          Defines how the request/response is displayed. Options are "split" or "float".
+        '';
+
+    split_direction =
+      defaultNullOpts.mkEnumFirstDefault
+        [
+          "vertical"
+          "horizontal"
+        ]
+        ''
+          Defines the direction of the split window. Options are "vertical" or "horizontal".
+        '';
+
+    default_view =
+      defaultNullOpts.mkEnumFirstDefault
+        [
+          "body"
+          "headers"
+          "headers_body"
+        ]
+        ''
+          Sets the default view mode for responses. Options are "body", "headers", or "headers_body".
+        '';
+
+    default_env = defaultNullOpts.mkStr "dev" ''
+      Sets the default environment for requests. Common options are "dev", "test", or "prod".
+    '';
+
+    debug = defaultNullOpts.mkBool false ''
+      Enables or disables debug mode.
+    '';
+
+    contenttypes =
+      defaultNullOpts.mkAttrsOf types.anything
+        {
+          "application/json" = {
+            ft = "json";
+            formatter = [
+              "jq"
+              "."
+            ];
+            pathresolver.__raw = "require('kulala.parser.jsonpath').parse";
+          };
+          "application/xml" = {
+            ft = "xml";
+            formatter = [
+              "xmllint"
+              "--format"
+              "-"
+            ];
+            pathresolver = [
+              "xmllint"
+              "--xpath"
+              "{{path}}"
+              "-"
+            ];
+          };
+          "text/html" = {
+            ft = "html";
+            formatter = [
+              "xmllint"
+              "--format"
+              "--html"
+              "-"
+            ];
+            pathresolver = [ ];
+          };
+        }
+        ''
+          Defines formatters and path resolvers for specific content types.
+        '';
+
+    show_icons =
+      defaultNullOpts.mkEnumFirstDefault
+        [
+          "on_request"
+          "above_request"
+          "below_request"
+        ]
+        ''
+          Determines where icons are displayed relative to requests. Options are "on_request", "above_request", "below_request", or "{__raw = "nil";}" to disable.
```

```suggestion
          Determines where icons are displayed relative to requests. Options are "on_request", "above_request", "below_request", or `{__raw = "nil";}` to disable.
```

Better without quotes, and instead marked as `` `code` ``? Also begs the question: whether the others should be `` `"string_val"` ``?

---

I like this better! Modified to code and added ``"string_val"``